src/views/RecIM/components/List/Listing/index.jsx

```diff
@@ -0,0 +1,358 @@
+    };
+    loadUserInfo();
+    loadAvatar();
+  }, [participant.Username]);
+
+  const handleParticipantOptions = (event) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleMakeCoCaptain = async () => {
+    let editedParticipant = {
+      Username: participant.Username,
+      RoleTypeID: 4,
+    }; // Role 4 is co-captain
+
+    await editTeamParticipant(parseInt(teamID), editedParticipant); // Role 4 is co-captain
+    handleClose();
+  };
+
+  const handleRemoveFromTeam = async () => {
+    let editedParticipant = {
+      Username: participant.Username,
+      RoleTypeID: 6,
+    }; // Role 6 is inactive
+
+    await editTeamParticipant(parseInt(teamID), editedParticipant);
+    handleClose();
+  };
+
+  if (!participant) return null;
+  return (
+    // first ListItem is used only for paddings/margins
+    // second ListItem (nested inside) is used to layout avatar and secondaryAction
+    <ListItem key={participant.Username} className={styles.listingWrapper}>
+      <ListItem
+        secondaryAction={
+          minimal ? (
+            <IconButton edge="end" onClick={() => callbackFunction(participant.Username)}>
+              <ClearIcon />
+            </IconButton>
+          ) : (
+            showParticipantOptions && (
+              <IconButton edge="end" onClick={handleParticipantOptions}>
+                <MoreHorizIcon />
+              </IconButton>
+            )
+          )
+        }
+        disablePadding
+      >
+        <ListItemButton to={`/profile/${participant.Username}`} className={styles.listing}>
+          <ListItemAvatar>
+            <Avatar
+              src={`data:image/jpg;base64,${avatar}`}
+              className={minimal ? styles.avatarSmall : styles.avatar}
+              variant="rounded"
+            ></Avatar>
+          </ListItemAvatar>
+          <ListItemText primary={name} secondary={participant.Role} />
+        </ListItemButton>
+        {showParticipantOptions && (
+          <Menu open={moreOptionsOpen} onClose={handleClose} anchorEl={anchorEl}>
+            <MenuItem dense onClick={handleMakeCoCaptain} divider>
+              Make co-captain
+            </MenuItem>
+            <MenuItem dense onClick={handleRemoveFromTeam} className={styles.redButton}>
+              Remove from team
+            </MenuItem>
+          </Menu>
+        )}
+      </ListItem>
+    </ListItem>
+  );
+};
+
+const MatchListing = ({ match, activityID }) => {
+  return (
+    <ListItem key={match.ID} className={styles.listingWrapper}>
+      <ListItemButton
+        component={Link}
+        to={`/recim/activity/${activityID}/match/${match.ID}`}
+        className={styles.listing}
+      >
+        <Grid container>
+          <Grid item>
+            {match.Team[0]?.Name ?? <i>TBD</i>} vs. {match.Team[1]?.Name ?? <i>TBD</i>}
+          </Grid>
+        </Grid>
+      </ListItemButton>
+    </ListItem>
+  );
+};
+
+export { ActivityListing, TeamListing, ParticipantListing, MatchListing };
```

Comment:
I feel like refactoring the Listing as a shell component might be difficult due to the intricacies of each listing component with each listing having a completely different grid layout to another one. The only one that is remotely similar is Participant and Team, but even within Team there are all the record types on the right side. Perhaps we could create a new file for each but it would make it look incredibly bloated in terms of file numbers. @EjPlatzer is this more of a stylistic change or is there a specific reason you want them in separate files?

Reply:
> The reason we did this initially was so that they could share styling and common features, but now that they have become more complex, I think we may be able to factor these out and use `Listing` as a shell component

They aren't actually sharing any code in the current iteration. If you plan to factor out repeated code in the future, that should be done via a re-usable component.

Currently, these are just a bunch of completely independent components sharing the file. The only thing they have somewhat in common is imports (but those don't exist in production builds so it doesn't make any sense to group components based on shared imports, otherwise we'd just write everything in the same file).

> I feel like refactoring the Listing as a shell component might be difficult due to the intricacies of each listing component with each listing having a completely different grid layout to another one. The only one that is remotely similar is Participant and Team, but even within Team there are all the record types on the right side.

This is, in my mind, proof that they should not be in the same file. IF they don't have anything in common beyond "being a list", why are they in the same file? And if they do have something in common, that thing in common should be extracted out to a re-usable component that also has it's own file.

> Perhaps we could create a new file for each but it would make it look incredibly bloated in terms of file numbers. @EjPlatzer is this more of a stylistic change or is there a specific reason you want them in separate files?

This is a somewhat fundamental code quality concern, so I wouldn't describe it as stylistic necessarily. You're thinking of this backward. Files exist so that we can keep them each small and self-contained. What you currently have is an extraordinarily bloated file (ideally, every file would have <100 lines of code ignoring imports). We use files to breakup a codebase into small, self-contained chunks that are easy to read and fully understand. So saying that breaking a bloated file into multiple files would cause "bloat" is exactly opposite the whole point of organizing source code into files.

Also, it's a react convention that each component should have it's own file. This is not a hard and fast rule, but it is a well-established convention and the exceptions (which mostly include extremely simple "helper" components) don't apply to this case.

Sorry if I'm coming across as harsh, that is not my intent, I just want to make sure I'm being clear.

Reply:
No I get it, makes sense to me! 👍🏼